src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -998,169 +998,19 @@ impl LinkCollector<'_, '_> {
-                            path_str,
-                            disambiguator,
-                            dox,
-                            link_range,
-                            candidates.into_iter().filter_map(|res| res.err()).collect(),
-                        );
-                        // this could just be a normal link
-                        return;
-                    }
-
-                    if len == 1 {
-                        candidates.into_iter().filter_map(|res| res.ok()).next().unwrap()
-                    } else if len == 2 && is_derive_trait_collision(&candidates) {
-                        candidates.type_ns.unwrap()
-                    } else {
-                        if is_derive_trait_collision(&candidates) {
-                            candidates.macro_ns = Err(ResolutionFailure::Dummy);
-                        }
-                        // If we're reporting an ambiguity, don't mention the namespaces that failed
-                        let candidates =
-                            candidates.map(|candidate| candidate.ok().map(|(res, _)| res));
-                        ambiguity_error(
-                            cx,
-                            &item,
-                            path_str,
-                            dox,
-                            link_range,
-                            candidates.present_items().collect(),
-                        );
-                        return;
-                    }
-                }
-                Some(MacroNS) => {
-                    match self.macro_resolve(path_str, base_node) {
-                        Ok(res) => (res, extra_fragment),
-                        Err(mut kind) => {
-                            // `macro_resolve` only looks in the macro namespace. Try to give a better error if possible.
-                            for &ns in &[TypeNS, ValueNS] {
-                                if let Some(res) = self.check_full_res(
-                                    ns,
-                                    path_str,
-                                    base_node,
-                                    &current_item,
-                                    &extra_fragment,
-                                ) {
-                                    kind = ResolutionFailure::WrongNamespace(res, MacroNS);
-                                    break;
-                                }
-                            }
-                            resolution_failure(
-                                self,
-                                &item,
-                                path_str,
-                                disambiguator,
-                                dox,
-                                link_range,
-                                smallvec![kind],
-                            );
-                            return;
-                        }
-                    }
-                }
+            match self.resolve_with_disambiguator(
```

Comment:
This is just `?`

Reply:
This returns `()`, not `Option`. So it can't use `?`.